File: LeetCode/1358/python/lc1358_test.py
import unittest

from lc1358 import Solution

class Test_numberOfSubstrings(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.numberOfSubstrings("abcabc"), 10)
        self.assertEqual(s.numberOfSubstrings("aaacb"), 3)
        self.assertEqual(s.numberOfSubstrings("abc"), 1)
        self.assertEqual(s.numberOfSubstrings("abb"), 0)

if __name__ == "__main__":
    unittest.main()

File: LeetCode/1358/golang/lc1358_test.go
package lc1358

import "testing"

func Test_numberOfSubstrings(t *testing.T) {
	testcases := []struct {
		s        string
		expected int
	}{
		{"abcabc", 10},
		{"aaacb", 3},
		{"abc", 1},
	}

	for i, tc := range testcases {
		actual := numberOfSubstrings(tc.s)
		if actual != tc.expected {
			t.Errorf("Testcase numberOfSubstrings#%02d (%s) failed: want %d, got %d",
				i+1, tc.s, tc.expected, actual)
		}
	}
}
